docs/content/3.middleware/2.guides/9.kubernetes-probes.md

```diff
@@ -0,0 +1,132 @@
+# Kubernetes Probes
+
+Alokai Cloud customers' middleware and frontend apps are deployed in Kubernetes. This document explains the Kubernetes mechanisms used by Alokai Cloud to ensure that customers' applications are starting, running and exiting correctly. Implementing your application so that it works in accordance with those mechanisms ensures better detctability of issues and lower error rates of your deployment.
+
+In order for some of those mechanisms to work, you - as the application developer - need to ensure certain REST endpoints exist in your application and that they respond with the correct status code. The sections below explain how to implement those endpoints.
+
+## Liveness probes
+
+### The purpose of liveness probes
+
+Without any kind of probes set up for your application, the only failure scenario where your app will be considered as not working correctly is if the process exits on startup or any time during the application runtime. For example, if you run `npm start` but the application secrets are missing from the environment, most applications' processes will immediately exit. The operating system could also kill the process due to lack of memory. 
+
+However consider the scenario where the process is running fine, but if you send a request to your application - it doesn't respond. In this case, Kubernetes still (wrongly) considers the application to be running correctly. After all - the process is still running.However, your users still can't use the application as their requests are not being responded to.
+
+Liveness probes help you avoid the above scenario, as they check if your application responds to a basic "hello world" HTTP request. This is checked every few seconds throughout your application's entire runtime. If a timeout or error response is received instead of a a success status code, the app is considered to be in a dead state (e.g. caught in an infinite loop due to an edge-case and developer error) and is restarted. 
```

Comment:
How about:
"Liveness probes can help you recover the application from a broken state when only restart can solve the problem. The application gets probed every few seconds to determine if the server can respond. If a timeout or error response is received instead of a success status code, the app is considered to be in a dead state (e.g. caught in an infinite loop due to an edge-case and developer error) and gets restarted."